crates/subspace-archiving/src/archiver.rs

```diff
@@ -185,6 +186,35 @@ pub struct NewArchivedSegment {
     pub object_mapping: Vec<PieceObjectMapping>,
 }
 
+impl NewArchivedSegment {
+    /// Returns all the object mappings in this archived segment.
+    pub fn global_object_mappings(&self) -> GlobalObjectMapping {
+        // Iterate through the object mapping vector for each piece
+        let objects = self
+            .object_mapping
+            .iter()
+            .zip(
+                self.segment_header
+                    .segment_index()
+                    .segment_piece_indexes_source_first(),
+            )
+            .flat_map(|(piece_mappings, piece_index)| {
+                // And then through each individual object mapping in the piece
+                piece_mappings
+                    .objects
+                    .iter()
+                    .map(move |piece_object| GlobalObject::V0 {
+                        hash: piece_object.hash(),
+                        piece_index,
+                        offset: piece_object.offset(),
+                    })
+            })
+            .collect();
+
+        GlobalObjectMapping { objects }
```

Comment:
This is fine, but I'm now thinking whether we need `GlobalObjectMapping` here. If not then we could return an iterator that lazily produces values with lower memory footprint.

Reply:
Sure, that would save approximately 8 bytes per mapping. And it fits better with a future feedback mechanism.

It's a little tricky to do this in a way that saves memory, because we don't want to hold a reference to `NewArchivedSegment`. Fixed in 0ea0b8ec41.